scripts/commerce.js

```diff
@@ -61,64 +61,173 @@ export const authPrivacyPolicyConsentSlot = {
+        const data = {};
+
+        // Process all JSONs in one pass
+        jsons.forEach((json) => {
+          if (json.data?.length) {
+            json.data.forEach(({ Key, Value }) => {
+              if (Key && Value !== undefined) {
+                data[Key] = Value;
+              }
+            });
           }
-          return [{}];
-        })
-        // process json from sources
-        .then(([json, oJson]) => {
-          const placeholders = {};
-
-          const allKeys = new Set([
-            ...(json.data?.map(({ Key }) => Key) || []),
-            ...(oJson?.data?.map(({ Key }) => Key) || []),
-          ]);
-
-          allKeys.forEach((Key) => {
-            if (!Key) return;
-            const keys = Key.split('.');
-            const originalValue = json.data?.find((item) => item.Key === Key)?.Value;
-            const overrideValue = oJson?.data?.find((item) => item.Key === Key)?.Value;
-            const finalValue = overrideValue ?? originalValue;
-            const lastKey = keys.pop();
-            const target = keys.reduce((obj, key) => {
-              obj[key] = obj[key] || {};
-              return obj[key];
-            }, placeholders);
-            target[lastKey] = finalValue;
+        });
+
+        // Early return if no valid data
+        if (Object.keys(data).length === 0) {
+          resolve({});
+          return;
+        }
+
+        // Convert data object to placeholders object with nested structure
+        const placeholders = {};
+
+        Object.entries(data).forEach(([Key, Value]) => {
+          const keys = Key.split('.');
+          const lastKey = keys.pop();
+          let target = placeholders;
+
+          // Navigate/create nested structure
+          keys.forEach((key) => {
+            target[key] = target[key] || {};
+            target = target[key];
           });
 
-          window.placeholders[prefix] = placeholders;
-          resolve(placeholders);
-        })
-        .catch((error) => {
-          console.error('error loading placeholders', error);
-          // error loading placeholders
-          window.placeholders[prefix] = {};
-          resolve(window.placeholders[prefix]);
+          // Set the final value
+          target[lastKey] = Value;
         });
-    });
-  }
-  return window.placeholders[`${prefix}`];
+
+        // Merge the new placeholders into the global merged object
+        Object.assign(window.placeholders._merged, placeholders);
+
+        resolve(placeholders);
+      })
+      .catch((error) => {
+        console.error('error loading placeholders', error);
+        // error loading placeholders
+        resolve({});
+      });
+  });
+
+  // Store the pending promise for this combination
+  window.placeholders._pending[cacheKey] = fetchPromise;
+
+  // Clean up pending promise when resolved
+  fetchPromise.finally(() => {
+    delete window.placeholders._pending[cacheKey];
+  });
```

Comment:
Do we care if a `fetchPromise` never resolves? Does this function handle that case or will the page break?

Reply:
The page won’t break because the function always returns a promise that eventually resolves, even if the fetch fails. Additionally, `Promise.all()` includes a `.catch()` that resolves with an empty object, and the main `fetchPromise` uses `.finally()` to ensure cleanup.

We could add a timeout, but the question is: how long is safe? Will it matter if one more request takes 30 seconds? In the end, it’s just another request in the background.

Reply:
> always returns a promise that eventually resolves

Even if the network req goes 100+ seconds?

Reply:
Like any other dependency, it will hang until it's done. If public `/config.json` hangs for 100+ seconds, we must wait for that response. Are you suggesting something different for placeholders?

Reply:
Fair :)